botorch/acquisition/proximal.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python3
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+r"""
+A wrapper around AquisitionFunctions to add proximal weighting of the
+acquisition function. Acquisition function is weighted via a squared exponential
+centered at the last training point, with varying lengthscales corresponding to
+'proximal_weights'. Can only be used with acqusition functions based on single
+batch models.
+
+Small values of proximal_weights corresponds to strong biasing towards recently
+observed points, which smoothes optimization with a small potential decrese in
+convergence rate.
+"""
+
+from __future__ import annotations
+
+import torch
+from botorch.acquisition import AcquisitionFunction
+from botorch.exceptions.errors import UnsupportedError
+from botorch.utils import t_batch_mode_transform
+from torch import Tensor
+from torch.distributions.multivariate_normal import _batch_mahalanobis
+from torch.nn import Module
+
+
+class ProximalAcquisitionFunction(AcquisitionFunction):
+    """A wrapper around AquisitionFunctions to add proximal weighting of the
+    acquisition function. Acquisition function is weighted via a squared exponential
+    centered at the last training point, with varying lengthscales corresponding to
+    'proximal_weights'. Can only be used with acqusition functions based on single
+    batch models.
+
+    Small values of proximal_weights corresponds to strong biasing towards recently
+    observed points, which smoothes optimization with a small potential decrese in
+    convergence rate.
```

Comment:
Same as above. You could also shorten the above version since it is given in detail here.